src/nvapi_gpu.cpp

```diff
@@ -342,8 +371,39 @@ extern "C" {
         if (!adapter->HasNvmlDevice())
             return HandleInvalidated(str::format(n, ": NVML available but current adapter is not NVML compatible"), alreadyLoggedHandleInvalidated);
 
+        nvmlGpuDynamicPstatesInfo_t gpuDynamicPstatesInfo;
+        auto result = adapter->GetNvmlDeviceDynamicPstatesInfo(&gpuDynamicPstatesInfo);
+        switch (result) {
+            case NVML_SUCCESS:
+                // nvmlGpuDynamicPstatesInfo_t also has `flags` but they are reserved for future use
+                pDynamicPstatesInfoEx->flags = 0;
+                // order of the first four utilization domains in NVML matches NVAPI
+                // to consider: should be avoid blindly copying domains 4-7 in case they don't match once introduced?
```

Comment:
Good question, I tend to say blindly copying is fine since chances are very high that nvapi will match nvml. But yeah, it's a small gamble.